src/compiler/separate_erasure_compiler.nit

```diff
@@ -516,12 +529,17 @@ class SeparateErasureCompilerVisitor
 			mtype = mtype.mtype
 			accept_null = "1"
 		end
+		if mtype isa MNotNullType then
+			mtype = mtype.mtype
+		end
 		if mtype isa MParameterType then
 			# Here we get the bound of the the formal type (eh, erasure...)
 			mtype = mtype.resolve_for(self.frame.mpropdef.mclassdef.bound_mtype, self.frame.mpropdef.mclassdef.bound_mtype, self.frame.mpropdef.mclassdef.mmodule, false)
 			if mtype isa MNullableType then
 				mtype = mtype.mtype
-				accept_null = "1"
+				if maybe_null then
```

Comment:
See my discussion about this with @xymus above. It’s a problem that occurs only since we have intersections.